lib/ecs_deploy/auto_scaler/auto_scaling_group_config.rb

```diff
@@ -184,14 +177,11 @@ def ec2_client
         )
       end
 
-      def instances(reload: false)
-        if reload || @instances.nil?
-          resp = client.describe_auto_scaling_groups({
-            auto_scaling_group_names: [name],
-          })
-          @instances = resp.auto_scaling_groups[0].instances
-        else
-          @instances
+      def describe_detachable_instances
+        client.describe_auto_scaling_groups({ auto_scaling_group_names: [name] }).auto_scaling_groups[0].instances.reject do |i|
+          # The lifecycle state of terminated instances becomes "Detaching", "Terminating", "Terminating:Wait", or "Terminating:Proceed",
+          # and we can't detach instances in such a state.
+          next true if i.lifecycle_state.start_with?("Terminating") || i.lifecycle_state == "Detaching"
```

Comment:
[Q] Why do you use `next true if #{condition}` instead of `#{condition}` ?

Reply:
Oh, I copied the following code and made some changes, but as you said, we can make it simpler. I'll fix it.

```ruby
if i.lifecycle_state.start_with?("Terminating")
  AutoScaler.error_logger.warn("#{log_prefix} The lifesycle state of #{i.instance_id} is \"#{i.lifecycle_state}\", so ignore it")
  next true
end
```

Reply:
Fixed in the commit https://github.com/reproio/ecs_deploy/pull/85/commits/20671cc9544ecd1d35dbb9dc2cde225f781ee2c7